src/main/script.ts

```diff
@@ -143,62 +114,30 @@ const parseScript = (input: string) => {
   })
 }
 
-const createCloseHandler = (browserWindow: BrowserWindow) => (code: number) => {
-  console.log(`k6 process exited with code ${code}`)
-  let channel = 'script:failed'
-  if (code === 0) {
-    channel = 'script:finished'
-  } else if (code === 105) {
-    channel = 'script:stopped'
-  }
-  browserWindow.webContents.send(channel)
-}
-
-const createChecksHandler =
-  (browserWindow: BrowserWindow) => (data: string) => {
-    // TODO: https://github.com/grafana/k6-studio/issues/277
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const checkData: K6Check[] = JSON.parse(data)
-    browserWindow.webContents.send('script:check', checkData)
-  }
-
-const createLogsHandler = (browserWindow: BrowserWindow) => (data: string) => {
-  // TODO: https://github.com/grafana/k6-studio/issues/277
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-  const logData: K6Log = JSON.parse(data)
-  browserWindow.webContents.send('script:log', logData)
-}
-
-export const archiveScript = (
+export const archiveScript = async (
   scriptPath: string,
   browserWindow: BrowserWindow
 ): Promise<string> => {
-  return new Promise((resolve, reject) => {
-    const k6Args = [
-      'archive',
+  try {
+    const client = new K6Client()
+
+    await client.archive({
       scriptPath,
-      '-O',
-      TEMP_K6_ARCHIVE_PATH,
-      '--log-format=json',
-    ]
-
-    spawnK6({
-      args: k6Args,
-      onStdErr: createLogsHandler(browserWindow),
-      onClose: (code) => {
-        if (code === 0) {
-          resolve(TEMP_K6_ARCHIVE_PATH)
-        } else {
-          browserWindow.webContents.send('script:failed')
-          reject(
-            new Error(
-              `Failed to create archive: k6 process exited with code ${code}`
-            )
-          )
-        }
-      },
+      outputPath: TEMP_K6_ARCHIVE_PATH,
     })
-  })
+
+    return TEMP_K6_ARCHIVE_PATH
+  } catch (error) {
+    browserWindow.webContents.send(ScriptHandler.Failed)
+
+    if (error instanceof ArchiveError) {
+      for (const log of error.stderr) {
+        browserWindow.webContents.send(ScriptHandler.Log, log)
+      }
```

Comment:
There's a slight difference in behavior here from the previous code: the previous code would send logs as they happened but this code will dump everything at the end of the process. I don't think that matters a whole lot since it's logging any errors that occurred while archiving the script.